src/main/java/com/netgrif/application/engine/workflow/service/TaskService.java

```diff
@@ -829,4 +830,19 @@ private EventOutcome addMessageToOutcome(Transition transition, EventType type,
         }
         return outcome;
     }
+
+    @Override
+    public SetDataEventOutcome getMainOutcome(Map<String, SetDataEventOutcome> outcomes, String taskId) {
+        SetDataEventOutcome mainOutcome;
+        String key = taskId;
+        if (!outcomes.containsKey(taskId)) {
+            Optional<String> optional = outcomes.keySet().stream().findFirst();
```

Comment:
does it matter which outcome is the main outcome if the taskId is not present?

If yes then the ordering of keys might not be guaranteed based on the used Map/Set implementation

Reply:
what if the map is empty?

Reply:
no, ordering doesnt matter, because in case with file endpoint there will be always exactly 1 task id. So map cant be empty.